docs/stateful-api-examples.md

```diff
@@ -0,0 +1,78 @@
+# Stateful API examples
```

Comment:
does this overlap with your other pr #23?

Reply:
I personally think good to separate them like this. That file is centered around understanding the concepts and using them in your code. Whereas this file (#24) is more centered around the CLI.